### PR TITLE
DOC: Various EA docs

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/20668

### CODE REVIEW

File: doc/source/extending.rst [hunk 1]
```diff
@@ -106,6 +113,24 @@ by some other storage type, like Python lists.
 See the `extension array source`_ for the interface definition. The docstrings
 and comments contain guidance for properly implementing the interface.
 
+We provide a test suite for ensuring that your extension arrays satisfy the expected
+behavior. To use the test-suite, you must provide several pytest fixtures and inherit
```
Comment (gfyoung) [inline doc/source/extending.rst]: "test-suite" or "test suite" (without the dash)?  You use both.

File: doc/source/extending.rst [hunk 2]
```diff
@@ -106,6 +113,24 @@ by some other storage type, like Python lists.
...
+from the base test class. The required fixtures are found in
+https://github.com/pandas-dev/pandas/blob/master/pandas/tests/extension/conftest.py.
+
+To use a test, subclass it
```
Comment (gfyoung) [inline doc/source/extending.rst]: Nit: do we want a colon at the end of this sentence?

**Comment by gfyoung** [comment]
BTW, I see some other docs changes (re: Python 2.7 support) that aren't quite pertinent to `ExtensionArray`, but they're minor enough that they should be fine.

File: pandas/core/arrays/base.py [hunk 3]
```diff
@@ -14,12 +14,15 @@ class ExtensionArray(object):
     with a custom type and will not attempt to coerce them to objects. They
     may be stored directly inside a :class:`DataFrame` or :class:`Series`.
 
+    .. versionadded:: 0.23.0
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/base.py]: Can you add a note about the experimental status in this file as well (eg in module docstring at line 1).

File: pandas/core/arrays/base.py [hunk 4]
```diff
@@ -35,6 +38,15 @@ class ExtensionArray(object):
     * _can_hold_na
     * _formatting_values
 
+    Some methods require casting the ExtensionArray to an ndarray of Python
+    objects, which may be expensive. When performance is a concern, we highly
+    recommend overriding the following methods.
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/base.py]: Maybe clarify that those have a default working implementation, but do a `self.astype(object)` ?

**Review by jorisvandenbossche — COMMENTED** [review]
Minor comments, looks good to me

**Comment by jreback** [comment]
thanks @TomAugspurger

**Pull request merged**